src/main/java/org/folio/helper/BindHelper.java

```diff
@@ -0,0 +1,158 @@
+import org.folio.rest.jaxrs.model.BindPiecesCollection;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.rest.jaxrs.model.ProcessingStatus;
+import org.folio.rest.jaxrs.model.ReceivingResult;
+import org.folio.rest.jaxrs.model.ReceivingResults;
+import org.folio.rest.jaxrs.model.Title;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class BindHelper extends CheckinReceivePiecesHelper<BindPiecesCollection> {
+
+  public BindHelper(BindPiecesCollection bindPiecesCollection,
+                    Map<String, String> okapiHeaders, Context ctx) {
+    super(okapiHeaders, ctx);
+    piecesByLineId = groupBindPieceByPoLineId(bindPiecesCollection);
+    if (logger.isDebugEnabled()) {
+      int poLineQty = piecesByLineId.size();
+      int piecesQty = StreamEx.ofValues(piecesByLineId)
+        .mapToInt(Map::size)
+        .sum();
+      logger.debug("{} piece records(s) are going to be bound for {} PO line(s)", piecesQty, poLineQty);
+    }
+  }
+
+  private Map<String, Map<String, BindPiecesCollection>> groupBindPieceByPoLineId(BindPiecesCollection bindPiecesCollection) {
+    String poLineId = bindPiecesCollection.getPoLineId();
+    Map<String, BindPiecesCollection> bindPieceMap = bindPiecesCollection.getBindPieceIds().stream()
+      .collect(Collectors.toMap(
+        bindPieceId -> bindPieceId,
+        bindPieceId -> bindPiecesCollection
+      ));
+
+    return Map.of(poLineId, bindPieceMap);
+  }
+
+  public Future<ReceivingResults> bindPieces(BindPiecesCollection bindPiecesCollection, RequestContext requestContext) {
+    return removeForbiddenEntities(requestContext)
+      .compose(vVoid -> processBindPieces(bindPiecesCollection, requestContext));
+  }
+
+  private Future<ReceivingResults> processBindPieces(BindPiecesCollection bindPiecesCollection, RequestContext requestContext) {
+    // 1. Get piece records from storage
+    return retrievePieceRecords(requestContext)
+      // 2. Update piece isBound flag
+      .map(this::updatePieceRecords)
+      // 4. Crate item for pieces with specific fields
+      .compose(piecesGroupedByPoLine -> createItemForPiece(piecesGroupedByPoLine, bindPiecesCollection, requestContext))
+      // 3. Update received piece records in the storage
+      .compose(piecesGroupedByPoLine -> storeUpdatedPieceRecords(piecesGroupedByPoLine, requestContext))
+      // 4. Update Title with new bind items
+      .map(piecesGroupedByPoLine -> updateTitleWithBindItems(piecesGroupedByPoLine, requestContext))
+      // 5. Return results to the client
+      .map(piecesGroupedByPoLine -> prepareResponseBody(piecesGroupedByPoLine, bindPiecesCollection));
+  }
+
+  private Map<String, List<Piece>> updatePieceRecords(Map<String, List<Piece>> piecesGroupedByPoLine) {
+    StreamEx.ofValues(piecesGroupedByPoLine)
+      .flatMap(List::stream)
+      .forEach(piece -> piece.setIsBound(true));
```

Comment:
most of all we need update other related piece fields with newly created item, I created separate story for this effort:
https://folio-org.atlassian.net/browse/MODORDERS-1112